### ISSUE TITLE
How to to train pixelsnail? 

### ISSUE THREAD

**Issue by preethiseshadri518** [rosinality/vq-vae-2-pytorch#68]
I am currently on the second stage to train the pixelsnail model. Do I train the top and bottom models separately, or do I first train the top first and use that checkpoint to train the bottom, since the model is hierarchical? Not sure I fully understand how this should be trained. I trained both separately and the loss was high/accuracy was low for the bottom model, so I was wondering if training should instead be sequential? Thanks!

**Comment by rosinality** [comment]
It is trained separately. You can refer to the VQ-VAE-2 paper for the overall procedures of training autoregressive prior models. Actually using the top model for the bottom model will not be beneficial for training the bottom model as encoded ground truth of top codes are used as additional input for the bottom model.

**Comment by preethiseshadri518** [comment]
Good to know I was doing it correctly. Thanks for confirming! From your experience, did you do anything differently to train the bottom model? The loss is high/accuracy is low and it doesn't seem to be improving over epochs.

**Comment by rosinality** [comment]
Default model setting in the repository is much smaller than the model represented in the paper. You can may try to increase it to match the paper.
